packages/api/src/beacon/routes/beacon/state.ts

```diff
@@ -373,6 +446,21 @@ export function getReqSerializers(): ReqSerializers<Api, ReqTypes> {
         query: {id: Schema.UintOrStringArray},
       },
     },
+
+    postStateValidatorBalances: {
+      writeReq: (state_id, ids) => ({
+        params: {state_id},
+        body: ids?.map((id) => (typeof id === "string" ? id : toU64Str(id))) || [],
+      }),
+      parseReq: ({params, body}) => [
+        params.state_id,
+        body?.map((id) => (typeof id === "string" && id.startsWith("0x") ? id : fromU64Str(id))),
```

Comment:
`id` is always a string? `fromU64Str` only accepts string.

Reply:
I don't think so, if we use string for validator index inside Lodestar it is likely unintended and should be fixed. And even though the type of `fromU64Str` states in only accepts strings (which imo it should), in practice it will handle numbers as type `number` just fine.

It only has to be a stringified number over the wire

[apis/beacon/states/validators.yaml#L115-L120](https://github.com/ethereum/beacon-APIs/blob/fc89ec5c5f0904a0057fc543f0eb4ddaa480f04e/apis/beacon/states/validators.yaml#L115-L120)
```yaml
ids:
 type: array
 uniqueItems: true
 items:
   description: "Either hex encoded public key (any bytes48 with 0x prefix) or validator index"
   type: string # <--
```

Reply:
Indices store strictly uses type `number`
https://github.com/ChainSafe/lodestar/blob/unstable/packages/validator/src/services/indices.ts#L42-L45